docs/adding-web-support-to-unimodules.md

```diff
@@ -0,0 +1,272 @@
+export default {
+  get name(): string {
+    return 'ExpoPrint';
+  },
+};
+```
+
+**iOS**
+
+```objc
+EX_EXPORT_MODULE(ExpoPrint);
+```
+
+**Android**
+
+```java
+@Override
+public String getName() {
+  return "ExpoPrint";
+}
+```
+
+### Constants
+
+Constants are defined as `get`ters after the `name` property.
+
+**Web**
+
+**`src/ExpoPrint.web.ts`**
+
+```ts
+export default {
+  get name(): string {
+    return 'ExpoPrint';
+  },
+  get EVENTS(): string {
+    return 'EVENTS';
+  },
+};
+```
+
+**iOS**
+
+```objc
+- (NSDictionary *)constantsToExport {
+    return @{
+        @"EVENTS": @"EVENTS"
+    };
+}
+```
+
+**Android**
+
+```java
+@Override
+public Map<String, Object> getConstants() {
+    final Map<String, Object> constants = new HashMap<>();
+    constants.put("EVENTS", "EVENTS");
+    return constants;
+}
+```
+
+## Methods
+
+Methods are all defined after the `name` property and any constants.
+
+**Web**
+
+**`src/ExpoPrint.web.ts`**
+
+```ts
+export default {
+  get name(): string {
+    return 'ExpoPrint';
+  },
+  get Orientation(): OrientationConstant {
+    return {
+      portrait: 'portrait',
+      landscape: 'landscape',
+    };
+  },
+  async printAsync(): Promise<void> {
+    return;
+  },
+};
+```
+
+**iOS**
+
+```objc
+EX_EXPORT_MODULE(ExpoPrint);
+
+- (NSDictionary *)constantsToExport
+{
+    return @{
+            @"Orientation": @{
+                @"portrait": @"portrait",
+                @"landscape": @"landscape"
+            }
+        };
+}
+
+EX_EXPORT_METHOD_AS(printAsync,
+                    printAsync:(EXPromiseResolveBlock)resolve
+                    reject:(EXPromiseRejectBlock)reject)
+{
+    resolve(nil);
+}
+```
+
+**Android**
+
+```java
+ @Override
+  public String getName() {
+    return "ExpoPrint";
+  }
+
+  @Override
+  public Map<String, Object> getConstants() {
+    return Collections.unmodifiableMap(new HashMap<String, Object>() {
+      {
+        put("Orientation", Collections.unmodifiableMap(new HashMap<String, Object>() {
+          {
+            put("portrait", "portrait");
+            put("landscape", "landscape");
+          }
+        }));
+      }
+    });
+  }
+
+  @ExpoMethod
+  public void printAsync(final Promise promise) {
+      promise.resolve(null);
+  }
+```
+
+## Return value
+
+Currently all methods are `async` and return a promise. This is until Turbo Modules are released, at which time we will add synchronous methods.
+
+**`src/ExpoPrint.web.ts`**
+
+```diff
+✅
++ export default {
++   //...
++   async getStringAsync(): Promise<string> {
++     return 'String';
++   },
++ };
+
+❌
+- export default {
+-   //...
+-   getStringAsync(): string {
+-     return 'String';
+-   },
+- };
+
+❌
+- export default {
+-   //...
+-   getStringAsync(): Promise<string> {
+-     return Promise.resolve('String');
+-   },
+- };
+```
+
+## Views
+
+[Here are some examples](https://github.com/expo/expo/tree/master/packages/expo/src/effects)
+
+- Create separate views for web
+- No `CSS`
```

Comment:
CSS is fine, it's external .css files that we want to avoid to keep the build process from getting more complicated